frontend/src/components/common/TextInput.js

```diff
@@ -0,0 +1,36 @@
+import React, {Component} from "react";
+import {observer} from "mobx-react";
+import PropTypes from "prop-types";
+
+import {randomString} from "../../common";
+
+@observer
+class TextInput extends Component {
+    constructor(props) {
+        super(props);
+        this._id = randomString();
+    }
+    render() {
+        const {onChange, label, value} = this.props;
+        return (
+            <>
```

Comment:
the TextInput is inconsistent with the IntegerInput; this isn't wrapped in a `<div className="form-group">` while the other is

Reply:
i made a LabelInput separately so moving to label and inputs into separate files.